huaweicloud/services/acceptance/workspace/data_source_huaweicloud_workspace_desktop_tags_filter_test.go

```diff
@@ -0,0 +1,268 @@
+func TestAccDesktopTagsFilterDataSource_basic(t *testing.T) {
+	var (
+		rName = acceptance.RandomAccResourceNameWithDash()
+
+		dcName = "data.huaweicloud_workspace_desktop_tags_filter.test"
+		dc     = acceptance.InitDataSourceCheck(dcName)
+
+		filterByWithoutTags   = "data.huaweicloud_workspace_desktop_tags_filter.filter_by_without_tags"
+		dcFilterByWithoutTags = acceptance.InitDataSourceCheck(filterByWithoutTags)
+
+		filterByAllTags   = "data.huaweicloud_workspace_desktop_tags_filter.filter_by_all_tags"
+		dcFilterByAllTags = acceptance.InitDataSourceCheck(filterByAllTags)
+
+		filterByAnyTags   = "data.huaweicloud_workspace_desktop_tags_filter.filter_by_any_tags"
+		dcFilterByAnyTags = acceptance.InitDataSourceCheck(filterByAnyTags)
+
+		filterByWithoutAllTags   = "data.huaweicloud_workspace_desktop_tags_filter.filter_by_without_all_tags"
+		dcFilterByWithoutAllTags = acceptance.InitDataSourceCheck(filterByWithoutAllTags)
+
+		filterWithoutAnyTags   = "data.huaweicloud_workspace_desktop_tags_filter.filter_by_without_any_tags"
+		dcFilterWithoutAnyTags = acceptance.InitDataSourceCheck(filterWithoutAnyTags)
+	)
+
+	resource.Test(t, resource.TestCase{
+		PreCheck: func() {
+			acceptance.TestAccPreCheck(t)
+		},
+		ProviderFactories: acceptance.TestAccProviderFactories,
+		Steps: []resource.TestStep{
+			{
+				Config: testAccDesktopTagsFilterDataSource_basic(rName),
+				Check: resource.ComposeTestCheckFunc(
+					dc.CheckResourceExists(),
+					resource.TestCheckResourceAttrSet(dcName, "desktops.#"),
+					resource.TestCheckResourceAttrSet(dcName, "desktops.0.resource_id"),
+					resource.TestCheckResourceAttrSet(dcName, "desktops.0.resource_name"),
+					resource.TestCheckResourceAttrSet(dcName, "desktops.0.tags.#"),
+					// without tags
+					dcFilterByWithoutTags.CheckResourceExists(),
+					resource.TestCheckOutput("is_without_tags_result_useful", "true"),
+					// with all tags
+					dcFilterByAllTags.CheckResourceExists(),
+					resource.TestCheckResourceAttrSet(filterByAllTags, "desktops.#"),
+					resource.TestCheckOutput("is_all_tags_result_useful", "true"),
+					// with any tags
+					dcFilterByAnyTags.CheckResourceExists(),
+					resource.TestCheckResourceAttrSet(filterByAnyTags, "desktops.#"),
+					resource.TestCheckOutput("is_any_tags_result_useful", "true"),
+					// without all tags
+					dcFilterByWithoutAllTags.CheckResourceExists(),
+					resource.TestCheckOutput("is_without_all_tags_result_useful", "true"),
+					// without any tags
+					dcFilterWithoutAnyTags.CheckResourceExists(),
+					resource.TestCheckOutput("is_without_any_tags_result_useful", "true"),
+				),
+			},
+		},
+	})
+}
+
+func testAccDesktopTagsFilterDataSource_basic(rName string) string {
+	tagKey := "owner"
+	tagValue := "terraform"
+	return fmt.Sprintf(`
+%[1]s
+
+locals {
+  tag_key   = "%[2]s"
+  tag_value = "%[3]s"
+}
+
+data "huaweicloud_workspace_desktop_tags_filter" "test" {
+  depends_on = [
+	  huaweicloud_workspace_desktop.test
+  ]
+}
```

Comment:
```suggestion
data "huaweicloud_workspace_desktop_tags_filter" "test" {
  depends_on = [huaweicloud_workspace_desktop.test]
}
```
or
```suggestion
data "huaweicloud_workspace_desktop_tags_filter" "test" {
  depends_on = [
    huaweicloud_workspace_desktop.test
  ]
}
```
Check and modify others

Reply:
done